homeassistant/components/nexia/climate.py

```diff
@@ -0,0 +1,524 @@
+    ATTR_MIN_HUMIDITY,
+    ATTR_MIN_TEMP,
+    ATTR_PRESET_MODE,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    ATTR_TARGET_TEMP_STEP,
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    SUPPORT_AUX_HEAT,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_ENTITY_ID,
+    ATTR_TEMPERATURE,
+    STATE_OFF,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ATTR_AIRCLEANER_MODE,
+    ATTR_DEHUMIDIFY_SETPOINT,
+    ATTR_DEHUMIDIFY_SUPPORTED,
+    ATTR_HOLD_MODES,
+    ATTR_HUMIDIFY_SETPOINT,
+    ATTR_HUMIDIFY_SUPPORTED,
+    ATTR_SETPOINT_STATUS,
+    ATTR_THERMOSTAT_ID,
+    ATTR_ZONE_ID,
+    ATTR_ZONE_STATUS,
+    ATTRIBUTION,
+    DATA_NEXIA,
+    DOMAIN,
+    MANUFACTURER,
+    NEXIA_DEVICE,
+    UPDATE_COORDINATOR,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_SET_AIRCLEANER_MODE = "set_aircleaner_mode"
+SERVICE_SET_HUMIDIFY_SETPOINT = "set_humidify_setpoint"
+
+SET_FAN_MIN_ON_TIME_SCHEMA = vol.Schema(
+    {
+        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_AIRCLEANER_MODE): cv.string,
+    }
+)
+
+SET_HUMIDITY_SCHEMA = vol.Schema(
+    {
+        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_HUMIDITY): vol.All(
+            vol.Coerce(int), vol.Range(min=35, max=65)
+        ),
+    }
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up climate for a Nexia device."""
+
+    nexia_data = hass.data[DOMAIN][config_entry.entry_id][DATA_NEXIA]
+    nexia_home = nexia_data[NEXIA_DEVICE]
+    coordinator = nexia_data[UPDATE_COORDINATOR]
+
+    entities = []
+    for thermostat_id in nexia_home.get_thermostat_ids():
+        thermostat = nexia_home.get_thermostat_by_id(thermostat_id)
+        for zone_id in thermostat.get_zone_ids():
+            zone = thermostat.get_zone_by_id(zone_id)
+            entities.append(NexiaZone(coordinator, zone))
+
+    def humidify_set_service(service):
+        entity_id = service.data.get(ATTR_ENTITY_ID)
+        humidity = service.data.get(ATTR_HUMIDITY)
+        target_zones = []
+
+        for zone in entities:
+            if entity_id and zone.entity_id not in entity_id:
+                continue
+
+            if (
+                zone.thermostat.has_humidify_support()
+                and zone.thermostat not in target_zones
+            ):
+                target_zones.append(zone)
+
+        for zone in target_zones:
+            thermostat.set_humidify_setpoint(int(humidity) / 100.0)
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_SET_HUMIDIFY_SETPOINT,
+        humidify_set_service,
+        schema=SET_HUMIDITY_SCHEMA,
+    )
```

Comment:
Do you need custom entity selection criteria or would you be able to leverage [`platform.async_register_entity_service`](https://github.com/home-assistant/core/blob/dev/homeassistant/components/sonos/media_player.py#L233) here?

Reply:
This is is already large, I deleted the services. will figure this out in a new pr.

Reply:
services deleted for this initial pr